merlion/models/anomaly/dagmm.py

```diff
@@ -207,41 +207,46 @@ def train(
         return train_scores
 
     def train_multiple(
-        self, train_data: List[TimeSeries], anomaly_labels: List[TimeSeries] = None,
-        train_config=None, post_rule_train_config=None
+        self, multiple_train_data: List[TimeSeries], anomaly_labels: List[TimeSeries] = None,
+        train_config=dict(), post_rule_train_config=None
```

Comment:
Can you change the default value of `train_config` back to `None`? See [here](https://stackoverflow.com/questions/1132941/least-astonishment-and-the-mutable-default-argument) for why `train_config=dict()` can be a problem. The preferred pattern would be `train_config = {} if train_config is None else train_config`.  Looks good to me once this is addressed.

Reply:
😅  Spending time with other languages makes me forget some of the Python pitfalls.
I agree that it is safer to avoid this at all even if a mutation doesn't happen.
Returned the `None` back with additional if.